File: projetotcc-front/src/app/projeto/projeto.service.ts
import { Injectable } from '@angular/core';
import {CrudService} from '../generic/crud.service';
import {HttpClient} from '@angular/common/http';
import {environment} from '../../environments/environment';
import {Projeto} from '../model/projeto';
import {Observable} from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class ProjetoService extends CrudService<Projeto, number>{

  constructor(http: HttpClient) {
    super(environment.api + '/projeto', http);
  }

  deleteProjeto(id: number): Observable<void> {
    const url = `${this.getUrl()}/delete/${id}`;
    return this.http.delete<void>(url);
  }

}

File: projetotcc-back/src/main/java/br/edu/utfpr/pb/projetotcc/service/ProjetoService.java
package br.edu.utfpr.pb.projetotcc.service;

import br.edu.utfpr.pb.projetotcc.model.Projeto;

import java.util.List;

public interface ProjetoService extends CrudService<Projeto, Long>{

    List<Projeto> findProjetoByMembrosId(Long membroId);

    void deleteProjeto(Long projetoId) throws Exception;
}
